go/vt/schema/online_ddl.go

```diff
@@ -68,42 +76,126 @@ func validateWalk(node sqlparser.SQLNode, allowForeignKeys bool) (kontinue bool,
 	return false, nil
 }
 
-// OnlineDDLStatus is an indicator to a online DDL status
-type OnlineDDLStatus string
-
-const (
-	OnlineDDLStatusRequested OnlineDDLStatus = "requested"
-	OnlineDDLStatusCancelled OnlineDDLStatus = "cancelled"
-	OnlineDDLStatusQueued    OnlineDDLStatus = "queued"
-	OnlineDDLStatusReady     OnlineDDLStatus = "ready"
-	OnlineDDLStatusRunning   OnlineDDLStatus = "running"
-	OnlineDDLStatusComplete  OnlineDDLStatus = "complete"
-	OnlineDDLStatusFailed    OnlineDDLStatus = "failed"
-)
-
 // OnlineDDL encapsulates the relevant information in an online schema change request
 type OnlineDDL struct {
-	Keyspace string      `json:"keyspace,omitempty"`
-	Table    string      `json:"table,omitempty"`
-	Schema   string      `json:"schema,omitempty"`
-	SQL      string      `json:"sql,omitempty"`
-	UUID     string      `json:"uuid,omitempty"`
-	Strategy DDLStrategy `json:"strategy,omitempty"`
-	Options  string      `json:"options,omitempty"`
+	// m is used to provide concurrent access to ReadyToComplete logic. Callers
+	// should use the IsReadyToComplete and MarkReadyToComplete methods to
+	// safely modify the value of that field rather than accessing it through
+	// the embedded protobuf message directly.
+	m sync.Mutex
+	*tabletmanagerdatapb.OnlineDDL
```

Comment:
Should we use `tabletmanagerdatapb.OnlineDDL` by value and not by pointer? This will be more inline with the current usage. Again, I'm concerned about copying `OnlineDDL` values: with `*tabletmanagerdatapb.OnlineDDL` we risk two instances of `OnlineDDL` sharing data. Maybe there's nothing in the code to worry about, and if all the tests pass then I guess everything's fine. But the original design was to share nothing, and I'm concerned we might miss a leak.